clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -256,11 +256,13 @@ public void poll(long timeout, long now, PollCondition pollCondition) {
      * nor will it execute any delayed tasks.
      */
     public void pollNoWakeup() {
-        disableWakeups();
-        try {
-            poll(0, time.milliseconds(), null);
-        } finally {
-            enableWakeups();
+        synchronized (this) {
```

Comment:
Yes, @ijuma is right. We were trying not to hold the lock when `poll()` is called. This is quite tricky. Seems like the `wakeupDisabledCount` variable is a bit problematic. I wonder if we should just pass down a flag in `poll()` indicating whether or not wakeups should be triggered?